oi/type_graph/Types.cpp

```diff
@@ -29,8 +29,8 @@ namespace type_graph {
 OI_TYPE_LIST
 #undef X
 
-std::string Primitive::name() const {
-  switch (kind_) {
+std::string Primitive::getName(Kind kind) {
```

Comment:
If the overarching `name()` call returns `std::string_view` this shouldn't need to be specialised or ever stored on the heap.

Reply:
I never got around to deciding whether I thought it was better to return a `std::string` or `std::string_view` so I'm just going to merge as-in for now, but this is something that we could explore in the future.